lib/topological_inventory/orchestrator/worker.rb

```diff
@@ -70,6 +71,11 @@ def each_source
 
             next unless (collector_definition = collector_definitions[source_type["name"]])
 
+            application_types = get_and_parse(sources_api_url_for("sources", source["id"], "application_types"), tenant)
+
+            enabled_applications = application_types&.dig("data")&.map { |app_type| app_type["name"] } || []
```

Comment:
Will `application_types` ever be `nil`?

Reply:
Ideally that's correct, `get_and_parse` does return `nil` on a `RestClient::NotFound` exception though not sure if that's something we need to handle here

Reply:
I could just `next if application_types.nil?` and avoid the `&. &.`, if it is non-nil then it should have the proper `"data"` key, wyt?

Reply:
I think the only time we would get a `RestClient::NotFound` is if the source doesn't exist and [we already account for that](https://github.com/ManageIQ/topological_inventory-orchestrator/pull/37/files#diff-4743ee12ee7e621468f2e6590de994efR68)

Reply:
:+1: done